File: Android/server_feedback/src/main/java/ru/po_znaika/server_feedback/ServerCacheFeedback.java
package ru.po_znaika.server_feedback;

import android.content.Context;

import java.util.Date;

import ru.po_znaika.database.diary.DiaryDatabase;

/**
 * Implements feedback with server with caching results in Diary database
 */
public class ServerCacheFeedback implements IServerFeedback
{
    //private static final String ServerAddressUri = "";

    public ServerCacheFeedback(Context context)
    {
        m_diaryDatabase = new DiaryDatabase(context);
    }

    public boolean reportExerciseResult(int exerciseId, int score)
    {
        //final boolean IsDataSended = sendDataToServer(exerciseId, score);

        return saveInLocalCache(exerciseId, score, false);
    }

    private boolean saveInLocalCache(int exerciseId, int score, boolean isSentToServer)
    {
        return m_diaryDatabase.insertExerciseScore(new Date(System.currentTimeMillis()), exerciseId, score,
                isSentToServer) != ru.po_znaika.database.DatabaseConstant.InvalidDatabaseIndex;
    }

    private DiaryDatabase m_diaryDatabase;
}

File: Windows/ServerFeedback/ServerCacheFeedback.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Net;
using ru.po_znaika.database.Diary;

namespace ru.po_znaika.server_feedback
{
    public class ServerCacheFeedback : IServerFeedback
    {
        public ServerCacheFeedback(string diaryDatabasePath, string serverAddressUri)
        {
            m_httpRequest = WebRequest.Create(serverAddressUri) as HttpWebRequest;
            m_diaryDatabase = new DiaryDatabase(diaryDatabasePath, false);
        }

        public void SendCachedDataToServer()
        {
        }

        public void SyncDiaryDataWithServer()
        {
        }
        
        public bool ReportExerciseResult(int exerciseId, int score)
        {
            bool isDataSended = SendDataToServer(exerciseId, score);

            return SaveInLocalCache(exerciseId, score, isDataSended);
        }

        private bool SaveInLocalCache(int exerciseId, int sccore, bool isSentToServer)
        {
            return m_diaryDatabase.InsertExerciseScore(DateTime.Now, exerciseId, sccore, isSentToServer) != ru.po_znaika.database.Constant.InvalidDatabaseIndex;
        }

        private bool SendDataToServer(int exerciseId, int score)
        {
            return false;
        }

        private bool SendDataToServer(IEnumerable<ExerciseDiaryInfo> records)
        {
            return false;
        }

        private HttpWebRequest m_httpRequest;
        private DiaryDatabase m_diaryDatabase;
    }
}
